nkms/client.py

```diff
@@ -200,7 +241,13 @@ def open(self, pubkey=None, path=None, mode='r', fd=None, algorithm=None):
 
         If pubkey is not set, we're working on our own files.
         """
-        pass
+        file_path = fd or path
+        try:
+            with open(file_path, mode=mode) as f:
+                enc_data = f.read()
```

Comment:
With open, it's better to decrypt file as a stream (gradually), so that we're working with the resulting file descriptor as if we opened an unencrypted file. E.g.:
```python
f = client.open(pub, '/hello/world.txt')
x = f.read(10)
...
f.close
```
or even:
```python
with client.open(...) as f:
    do_something_with(f)
```

Reply:
So, should I implement a file-like interface for this method and not have it return the unencrypted data?

My first thought was to have it just decrypt the entire file at once for testing, then when we start focusing more on the cryptographic API to place the NaCl nonce in the NuCypher header so we can decrypt it as a stream.